File: inspect_excel.js
const XLSX = require('xlsx');

const filename = 'Checkpoint Daily Portfolio History 111825.xlsx';
console.log(`Reading ${filename}...`);

try {
    const workbook = XLSX.readFile(filename);
    const sheetNames = workbook.SheetNames;
    console.log('Sheets:', sheetNames);

    const histSheetName = sheetNames.find(n =>
        n.toLowerCase().includes('itd history') ||
        n.toLowerCase().includes('daily portfolio') ||
        n.toLowerCase().includes('history portfolio')
    );

    if (histSheetName) {
        console.log(`Found History Sheet: ${histSheetName}`);
        const sheet = workbook.Sheets[histSheetName];
        const data = XLSX.utils.sheet_to_json(sheet);

        if (data.length > 0) {
            console.log('First row keys:', Object.keys(data[0]));

            // Find columns
            const firstRow = data[0];
            const dateCol = Object.keys(firstRow).find(k => k.toLowerCase().includes('date') || k.toLowerCase().includes('as of'));
            const tickerCol = Object.keys(firstRow).find(k => k.toLowerCase().includes('ticker') || k.toLowerCase().includes('symbol'));

            console.log(`Date Col: ${dateCol}, Ticker Col: ${tickerCol}`);

            if (dateCol && tickerCol) {
                const ironRows = data.filter(r => r[tickerCol] === 'IRON');
                console.log(`Found ${ironRows.length} rows for IRON`);

                if (ironRows.length > 0) {
                    // Sort by date
                    ironRows.sort((a, b) => a[dateCol] - b[dateCol]);

                    const first = ironRows[0];
                    const last = ironRows[ironRows.length - 1];

                    console.log('First IRON entry:', first[dateCol], parseDate(first[dateCol]));
                    console.log('Last IRON entry:', last[dateCol], parseDate(last[dateCol]));
                }

                // Check global max date
                let maxDateVal = 0;
                data.forEach(r => {
                    if (r[dateCol] > maxDateVal) maxDateVal = r[dateCol];
                });
                console.log('Global Max Date:', maxDateVal, parseDate(maxDateVal));
            }
        }
    } else {
        console.log('History sheet not found');
    }

} catch (e) {
    console.error('Error:', e.message);
}

function parseDate(excelDate) {
    if (typeof excelDate === 'number') {
        const date = new Date(Math.round((excelDate - 25569) * 86400 * 1000));
        return date.toISOString().split('T')[0];
    }
    return excelDate;
}

File: inspect_excel.py
import pandas as pd
import os

file_path = 'Checkpoint Daily Portfolio History 111825.xlsx'

if not os.path.exists(file_path):
    print(f"File not found: {file_path}")
    exit()

xl = pd.ExcelFile(file_path)
print(f"Sheet names: {xl.sheet_names}")

# Find history sheet
hist_sheet = next((s for s in xl.sheet_names if 'history' in s.lower() or 'daily' in s.lower()), None)

if hist_sheet:
    print(f"Reading sheet: {hist_sheet}")
    df = pd.read_excel(file_path, sheet_name=hist_sheet)
    print("Columns:", df.columns.tolist())
    
    # Check for Date column
    date_col = next((c for c in df.columns if 'date' in str(c).lower() or 'as of' in str(c).lower()), None)
    ticker_col = next((c for c in df.columns if 'ticker' in str(c).lower() or 'symbol' in str(c).lower()), None)
    
    if date_col and ticker_col:
        print(f"Date Column: {date_col}")
        print(f"Ticker Column: {ticker_col}")
        
        # Check IRON entries
        iron_df = df[df[ticker_col] == 'IRON']
        if not iron_df.empty:
            print(f"Found {len(iron_df)} rows for IRON")
            print("First 5 rows:")
            print(iron_df.head())
            print("Last 5 rows:")
            print(iron_df.tail())
            
            # Check max date
            print(f"Max Date in File: {df[date_col].max()}")
            print(f"Max Date for IRON: {iron_df[date_col].max()}")
        else:
            print("No entries found for IRON")
    else:
        print("Could not identify Date or Ticker columns")
else:
    print("History sheet not found")
